gdx/src/com/badlogic/gdx/math/collision/OrientedBoundingBox.java

```diff
@@ -0,0 +1,142 @@
+/*******************************************************************************
+ * Copyright 2011 See AUTHORS file.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.math.collision;
+
+import com.badlogic.gdx.math.Matrix4;
+import com.badlogic.gdx.math.Vector3;
+
+import java.io.Serializable;
+
+public class OrientedBoundingBox implements Serializable {
+	private static final long serialVersionUID = 3864065514676250557L;
+
+	private final static Vector3 tmpVector = new Vector3();
+	private final static Matrix4 tmpMatrix = new Matrix4();
+
+	public final BoundingBox bounds = new BoundingBox();
```

Comment:
I'm a bit worried about performances issues because of lot of unnecessary matrix inverse calculation and also matrix multiplication (for corners). I think we should cache them.
BoundingBox requires explicit call to update method when min/max are changed manually. We did that to avoid breaking changes but i think we should have something cleaner for this new class : 

* bounds and transform as private fields
* getters with proper documentation saying that returned object shouldn't be modified manually
* setTransform updating inverse matrix
* setBounds updating corners
* set(bounds, transform) updating both inverse matrix and corners.

what do you think?

Reply:
Yeah, some of these crossed my mind but would be too much work for something that I have a certain urgency to have on libgdx.

I agree with your concerns, I will work on this before fixing the intersection.

Reply:
I tried something along those lines